bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -40,18 +41,20 @@
  * @author Mark Herwege - Added methods to find broadcast address(es)
  * @author Stefan Triller - Converted to OSGi service with primary ipv4 conf
  */
-@Component(configurationPid = "org.eclipse.smarthome.network", property = { "service.pid=org.eclipse.smarthome.network",
-        "service.config.description.uri=system:network", "service.config.label=Network Settings",
-        "service.config.category=system" })
+@Component(immediate = true, configurationPid = "org.eclipse.smarthome.network", property = {
```

Comment:
How is the `immediate` change related to the topic of this change?
A utility method does IMHO only be necessary if someone uses it.
Do you add some magic action that does need to be executed also if there is no consumer of this service?

Reply:
@maggu2810 This change was actually suggested by me. I was testing the paper UI configuration and noticed that the `modified` method of the `NetUtil` class was not called due to the fact that the service was not started yet. I was asking @kaikreuzer and @sjka if there is any reason to not start it immediately and they also said that there is no reason to not start it.

And since @mherwege was touching the code anyway I thought it won't hurt if he includes the change in this PR.

Reply:
@triller-telekom Thank you for the clarification.

Reply:
There are services that brings some functionality on its own without any consumers. This ones should be started also if no other one reference this service. This ones should use `immediate = true`.
There are other services that provides some function for consumers. They do not need to be activated if no one uses it. This ones does not need to be activated immediately.

Surely we could set immediate for stuff regardless if it is used or not. Perhaps we already know, it will be used, but then there is still no need to set it explicit, because the framework will activate it if used.

So, I don't understand why we should set immediate here? "It makes no difference because the core will depend on that service" shouldn't be an argument. Shouldn't a component declaration be independent of its special use case and shouldn't we leave it up to the framework to do it correctly?

Or do I miss the point here and modified needs to be called without any consumer available?

Reply:
If you are really sure that this service will always be started without the `immediate = true`, I am fine with removing this again.

@mherwege: Sorry about the noise, but could you please revert this?

Reply:
Seems we don't matter. Already merged.

Reply:
@triller-telekom @maggu2810 I missed this when doing the last changes requested by @Kai. Do you expect me to create a PR to revert this change?

Reply:
... and I didn't notice that it was missing - didn't mean to ignore it, sorry.
@mherwege, yes, please just create a small new PR for it - thanks!